### PR TITLE
Fix warnings in examples

### LINKED ISSUE #292: htest examples - don't use formula method
>  I wonder about that warning... Since it's known that we cannot retrieve the data object, is it necessary to print the warning? Better perhaps a footer like "Effect size obtained via t to d conversion" @mattansb thoughts?

But sometimes we can retrieve data:

```r
x <- t.test(mtcars$mpg, factor(mtcars$am))

effectsize::effectsize(x) # gives Cohen's d

effectsize::effectsize(x, type = "hedges_g")
```

does work without any approximation.

_Originally posted by @mattansb in https://github.com/easystats/report/issues/289#issuecomment-1273437204_

---

Therefore **I suggest using only vector method (not the formula method) in `report` examples of `htest`.**

At least until https://github.com/easystats/effectsize/pull/522 is resolved.

### PR DESCRIPTION
**Pull request by rempsyc**
Closes #292

Fix warnings in examples, update authorship, bump version, update cran comments

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -71,7 +71,8 @@ Suggests:
     rmarkdown,
     rstanarm,
     survival,
-    testthat
+    testthat,
+    spelling
```
Comment (IndrajeetPatil) [inline DESCRIPTION:75]: Please don't gain spelling as a dependency. We don't need it.

We install spelling in the GHA, and so we don't need to gain it as a dependency.

File: tests/spelling.R [hunk 2]
```diff
@@ -0,0 +1,3 @@
+if(requireNamespace('spelling', quietly = TRUE))
```
Comment (IndrajeetPatil) [inline tests/spelling.R:1]: Revert this entire file; we don't need it.

**Review by IndrajeetPatil — CHANGES_REQUESTED** [review]
Just need to revert spelling related changes, and this should be good. Thanks!

File: DESCRIPTION (hunk 1 above)
Comment (rempsyc) [reply DESCRIPTION:75]: Whoops, old habits sorry!

**Review by IndrajeetPatil — APPROVED** [review]

**Comment by IndrajeetPatil** [comment]
Thanks for the amazing work, @rempsyc!

**Pull request merged**